src/Liquid/AbstractBlock.php

```diff
@@ -111,14 +111,14 @@ public function parse(array &$tokens)
 	 */
 	protected function whitespaceHandler(&$token)
 	{
-		if (mb_strlen($token) > 2 && $token[2] === Liquid::get('WHITESPACE_CONTROL')) {
+		if (mb_strlen($token) > 2 && mb_substr($token, 2, 1) === Liquid::get('WHITESPACE_CONTROL')) {
 			$previousToken = end($this->nodelist);
 			if (is_string($previousToken)) {
 				$this->nodelist[key($this->nodelist)] = rtrim($previousToken);
 			}
 		}
 
-		self::$trimWhitespace = (mb_strlen($token) > 2 && substr($token, -3, 1) === Liquid::get('WHITESPACE_CONTROL'));
+		self::$trimWhitespace = (mb_strlen($token) > 2 && mb_substr($token, -3, 1) === Liquid::get('WHITESPACE_CONTROL'));
```

Comment:
What do you think if we just use `mb_substr` here and remove `mb_strlen` altogether?